### PR TITLE
Add honggfuzz to fuzzing engines

### PR DESCRIPTION
**Pull request by ammaraskar**
This allows benchmarks to be built with honggfuzz as long as you point `HONGGFUZZ_SRC` to the git repo with `make` ran in it. 

Additionally, this also integrates honggfuzz into the AB-comparison engine. The docker images for the runner and dispatcher will need to be rebuilt and pushed up as they now include honggfuzz's runtime dependencies.

### CODE REVIEW

**Comment by Dor1s** [comment]
//cc @robertswiecki to take a look as well

**Comment by robertswiecki** [comment]
lgtm, not sure what your strategy of managing the input corpus is though. The --covdir_all will save all de-duplicated (without those with overlapping coverage, basically a corpus minimization) as well as new ineresting inputs to this dir. If you're interested in inputs which cause new coverage you can use --covdir_new instead

**Comment by ammaraskar** [comment]
I think it's fine to use the minimized corpus (`--covdir_all`). For reference, with AFL we're currently just pulling the entire mutation queue, potentially including duplicates.

File: engine-comparison/dispatcher.sh [hunk 1]
```diff
@@ -132,6 +132,16 @@ download_engine() {
 
   . "${fengine_config}"
   case "${FUZZING_ENGINE}" in
+    honggfuzz)
+      if [[ ! -f "${HONGGFUZZ_SRC}/" ]]; then
```
Comment (morehouse) [inline engine-comparison/dispatcher.sh]: Should this be `-d`?

**Review by morehouse — APPROVED** [review]

**Pull request merged**